### PR TITLE
Replaced NodeInfo PubKey to NodeID

### PR DESCRIPTION
**Pull request by eeonevision**
* [x] Updated all relevant documentation in docs
* [x] Updated all code comments where relevant
* [x] Wrote tests
* [ ] Updated CHANGELOG.md

### CODE REVIEW

File: docs/specification/new-spec/p2p/peer.md [hunk 1]
```diff
@@ -83,7 +83,7 @@ The Tendermint Version Handshake allows the peers to exchange their NodeInfo:
 
 ```golang
 type NodeInfo struct {
-  PubKey     crypto.PubKey
+  NodeID    p2p.ID
```
Comment (melekes) [inline docs/specification/new-spec/p2p/peer.md]: just `ID`, no need for `Node` prefix

File: docs/specification/new-spec/p2p/peer.md [hunk 2]
```diff
@@ -95,7 +95,7 @@ type NodeInfo struct {
 ```
 
 The connection is disconnected if:
-- `peer.NodeInfo.PubKey != peer.PubKey`
+- `peer.NodeInfo.PubKey` is not equal to `peer.PubKey.Address()`
```
Comment (melekes) [inline docs/specification/new-spec/p2p/peer.md]: `peer.NodeInfo.ID is not equal to peerConn.ID`

**Comment by melekes** [comment]
Tests are failing:
```
benchmarks/codec_test.go:20:4:warning: unknown field PubKey in struct literal (gosimple)
```

```
# github.com/tendermint/tendermint/benchmarks
benchmarks/codec_test.go:20:10: unknown field 'PubKey' in struct literal of type p2p.NodeInfo
benchmarks/codec_test.go:46:9: unknown field 'PubKey' in struct literal of type p2p.NodeInfo
benchmarks/codec_test.go:66:9: unknown field 'PubKey' in struct literal of type p2p.NodeInfo
```

File: benchmarks/codec_test.go [hunk 3]
```diff
@@ -4,29 +4,29 @@ import (
...
+	crypto "github.com/tendermint/go-crypto"
+	wire "github.com/tendermint/go-wire"
 	proto "github.com/tendermint/tendermint/benchmarks/proto"
-	"github.com/tendermint/tendermint/p2p"
 	ctypes "github.com/tendermint/tendermint/rpc/core/types"
 )
 
 func BenchmarkEncodeStatusWire(b *testing.B) {
 	b.StopTimer()
-	cdc := amino.NewCodec()
```
Comment (melekes) [inline benchmarks/codec_test.go]: I think you forgot to `make get_vendor_deps`. This is the new version.

Comment (melekes) [reply benchmarks/codec_test.go]: It just got merged to develop. Sorry

**Comment by melekes** [comment]
Why did you close it?

**Comment by eeonevision** [comment]
I opened new pull request, because made rebase from latest changes and thought that my changes were broken.

**Comment by eeonevision** [comment]
https://github.com/tendermint/tendermint/pull/1443

It works :)
screen shot 2018-04-09 at 8 46 35 pm

**Pull request closed without merging**